### PR TITLE
Create a new vectortile role

### PR DESCRIPTION
**Pull request by pnorman**
This role is for tilekiln vector tile servers and it currently sets up and starts tilekiln and has the scripts necessary to import data.

It's getting long enough that I'd like a review

### CODE REVIEW

File: cookbooks/postgresql/resources/schema.rb
Comment (pnorman) [inline cookbooks/postgresql/resources/schema.rb:5]: todo: update year

File: cookbooks/postgresql/resources/schema.rb
Comment (pnorman) [inline cookbooks/postgresql/resources/schema.rb:30]: I'm not happy with the complexity of this. Schemas are per-DB and created by chef with access permissions. This is different than other resources. Tablespaces are clusterwide and created with chef while tables are per-db and not created by chef.

File: cookbooks/vectortile/README.md
Comment (pnorman) [inline cookbooks/vectortile/README.md:1]: rename

File: cookbooks/vectortile/attributes/default.rb
Comment (pnorman) [inline cookbooks/vectortile/attributes/default.rb:3]: Remove stray comment from testing

**Comment by pnorman** [comment]
ping @Firefishy

**Comment by tomhughes** [comment]
I'm probably better placed to review this... I'll try and get to it over the weekend.

File: cookbooks/vectortile/recipes/default.rb
Comment (tomhughes) [inline cookbooks/vectortile/recipes/default.rb:183]: Does this lock file dance achieve anything useful? Can this not just subscribe to `postgresql_database[tiles]` or `postgresql_schema[schema]` so that is happens when the database is created?

File: cookbooks/vectortile/recipes/default.rb
Comment (tomhughes) [inline cookbooks/vectortile/recipes/default.rb:205]: Can we add `sandbox true` here, maybe with some additional relaxations? What does `tilekiln serve` need to be able to do/access?

File: cookbooks/vectortile/recipes/default.rb
Comment (tomhughes) [inline cookbooks/vectortile/recipes/default.rb:203]: Why a blank line here?

File: cookbooks/vectortile/recipes/default.rb
Comment (tomhughes) [inline cookbooks/vectortile/recipes/default.rb:210]: I think this is redundant as all systemd services support restart.

File: roles/vectortile.rb
Comment (tomhughes) [inline roles/vectortile.rb:9]: Probably remove `jburgess` here?

**Review by tomhughes — CHANGES_REQUESTED** [review]

File: cookbooks/vectortile/templates/default/import-planet.erb
Comment (tomhughes) [reply cookbooks/vectortile/templates/default/import-planet.erb:17]: I think we used to and when we switched to `osm2psql-replicate` that was initialised manually so now the URL is in the database...

File: cookbooks/vectortile/templates/default/tilekiln-storage-update.erb
Comment (pnorman) [reply cookbooks/vectortile/templates/default/tilekiln-storage-update.erb]: This will come in a later PR. Removed for now

**Comment by pnorman** [comment]
I believe all issues have been fixed.

**Comment by pnorman** [comment]
Is it better to merge this in large chunks like this one, or wait till a machine is assigned and then merge it and assign the machine the appropriate role?

File: cookbooks/vectortile/templates/default/import-planet.erb
Comment (tomhughes) [inline cookbooks/vectortile/templates/default/import-planet.erb:24]: That looks like the wrong attribute? Also the comment is out of date now we're specifiying the server.

**Review by tomhughes — CHANGES_REQUESTED** [review]

**Comment by tomhughes** [comment]
> Is it better to merge this in large chunks like this one, or wait till a machine is assigned and then merge it and assign the machine the appropriate role?

I don't mind when it's merged but I'd rather not have it amended once it's been reviewed - better to merge it and make changes in additional PRs as it's easier to review additional changes than have to re-review a big change like this because it's been updated.

File: cookbooks/vectortile/templates/default/import-planet.erb
Comment (pnorman) [reply cookbooks/vectortile/templates/default/import-planet.erb:24]: It is the wrong attribute, I'll fix in a subsequent commit

**Comment by pnorman** [comment]
28a07fc3a5f501d61aa6d3a9191c0e4d94bc08a2 is ready for review.

File: cookbooks/tile/attributes/default.rb
Comment (tomhughes) [inline cookbooks/tile/attributes/default.rb:3]: Why is this changing the `tile` cookbook?

Comment (pnorman) [reply cookbooks/tile/attributes/default.rb:3]: Whoops - that's a mistake. I must have mixed up the many default.rb windows I had open.

File: cookbooks/tile/attributes/default.rb
Comment (tomhughes) [inline cookbooks/tile/attributes/default.rb:3]: This is still changed, and is causing a test failure.

**Review by tomhughes — CHANGES_REQUESTED** [review]
Can we just squash the commits together please - there doesn't seem to be any sort of structure to them other than the order in which they were made so it might as well be a single commit.

**Comment by pnorman** [comment]
I think everything is good here. I've rebased on master and I'm successfully keeping the DB up to date and generating vector tiles

File: cookbooks/vectortile/files/default/html/robots.txt
Comment (tomhughes) [inline cookbooks/vectortile/files/default/html/robots.txt:13]: Is this right for vector tiles?

File: cookbooks/vectortile/recipes/default.rb
Comment (tomhughes) [inline cookbooks/vectortile/recipes/default.rb:73]: This shouldn't be necessary now - the pinning is now handled in the `apt::default` recipe and includes osm2pgsql for Debian 12 hosts.

File: cookbooks/vectortile/recipes/default.rb
Comment (tomhughes) [inline cookbooks/vectortile/recipes/default.rb:75]: Was there a reason for using `apt_package` here?

File: cookbooks/vectortile/recipes/default.rb
Comment (tomhughes) [inline cookbooks/vectortile/recipes/default.rb:81]: If that comment isn't out of date already then it soon will be... Usually we specify the branch explicitly if we want to track a branch as the default it to track `HEAD` which is the upstream's default branch and at leats in theory subject to change.

**Review by tomhughes — CHANGES_REQUESTED** [review]
One thing to bear in mind is that we need to add the new users to the private repo before we merge this.

File: cookbooks/vectortile/files/default/html/robots.txt
Comment (pnorman) [reply cookbooks/vectortile/files/default/html/robots.txt:13]: Zoom levels 20+ don't exist for our raster or vector tiles. I assumed they were added to robots.txt because people were requesting them, and the same could happen with vector tiles.

File: cookbooks/vectortile/recipes/default.rb
Comment (pnorman) [reply cookbooks/vectortile/recipes/default.rb]: Fixed. I want to pin the versions like we do for osm-carto and such, but everything needs to be a bit more stable first.

**Comment by pnorman** [comment]
The test failures are in db-base, master, and slave and seem unrelated to this PR.

**Review by tomhughes — APPROVED** [review]
I think this is good enough to merge now...

**Pull request merged**
